src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -1113,24 +1112,24 @@ def forward(
             output_hidden_states=output_hidden_states,
             return_dict=return_dict,
         )
-        extractor_features = encoder_outputs.extracted_features
-        extractor_features = self.dropout_features(extractor_features)
+        extractor_features = self.dropout_features(encoder_outputs.extracted_features)
         transformer_features = encoder_outputs.last_hidden_state
 
+        # 1. quantize unmasked features
         quantized_extractor_features, transformer_features, negatives, num_vars, prob_perplexity = self.quantizer(
             extractor_features, transformer_features, encoder_outputs.mask_time_indices
         )
 
         # if a negative vector is identical to the positive (i.e. when codebook utilization is low),
         # its cosine similarity will be masked
-        neg_is_pos = (quantized_extractor_features == negatives).all(-1)
+        neg_is_pos = (transformer_features == negatives).all(-1)
 
-        quantized_extractor_features = quantized_extractor_features.unsqueeze(0)
-        target_states = torch.cat([quantized_extractor_features, negatives], dim=0)
-        logits = torch.cosine_similarity(transformer_features.float(), target_states.float(), dim=-1).type_as(
+        target_states = torch.cat([transformer_features[None, :], negatives], dim=0)
```

Comment:
Here the "`y`" (= `transformer_features`) should be used and not the "`x`"  (= `quantized_extractor_features`) I think